src/appimagetool.c

```diff
@@ -903,6 +908,33 @@ main (int argc, char *argv[])
                         printf("Will not guess update information since zsyncmake is missing\n");
                     }
                 }
+            } else if (github_repository != NULL && github_ref != NULL) {
+                printf("Running on GitHub Actions\n");
+                gchar *zsyncmake_path = g_find_program_in_path ("zsyncmake");
+                if (zsyncmake_path) {
+                    if (strstr(github_ref, "/pull/")) {
+                        printf("Will not calculate update information for GitHub because this is a pull request\n");
+                    } else {
+                        printf("Guessing update information based on $GITHUB_REPOSITORY=%s and $GITHUB_REF=%s\n", github_repository, github_ref);
+                        char buf[1024];
+                        gchar **parts = g_strsplit (github_repository, "/", 2);
+                        char* channel;
+                        if (github_ref != "" && strstr(github_ref, "/continuous/")) {
+                            channel = "latest";
+                        } else {
+                            channel = "continuous";
+                        }
+                        int is_zsync_write_success = snprintf(buf, 1024, "gh-releases-zsync|%s|%s|%s|%s*-%s.AppImage.zsync", parts[0], parts[1], channel, app_name_for_filename, arch);
```

Comment:
With arrays (i.e., `char ...[]` instead of `char*`), you can make use of the `sizeof` operator to get the amount of bytes contained. You can replace the `1024` with `sizeof(buf)` safely.

Otherwise, consider defining your buffer size as a `const` or `static const` and reference that instead of repeating the value more than once.

Also, 1024 might be too small. I'd suggest to orientate on the size of the update information section in the runtime's ELF header. Isn't that read somewhere anyway? If not, a new issue should be created.

Reply:
These buffers could be allocated dynamically via asprintf():

  if (asprintf(&updateinformation, ...) == -1) {
    abort();  // there's no point in trying to recover from a memory allocation failure.
  }